tests/app/main/test_services.py

```diff
@@ -715,18 +934,33 @@ def test_update_with_under_50_words_error(self, data_api_client):
 
     def test_update_non_existent_service_returns_404(self, data_api_client):
         data_api_client.get_service.return_value = None
-        res = self.client.post('/suppliers/services/1/edit/description')
+        res = self.client.post('/suppliers/frameworks/g-cloud-6/services/1/edit/description')
 
         assert res.status_code == 404
         self.assert_no_flashes()
+        assert data_api_client.update_service.called is False
 
     def test_update_non_existent_section_returns_404(self, data_api_client):
         data_api_client.get_service.return_value = self.empty_service
         res = self.client.post(
-            '/suppliers/services/1/edit/invalid_section'
+            '/suppliers/frameworks/g-cloud-6/services/1/edit/invalid_section'
         )
         assert res.status_code == 404
         self.assert_no_flashes()
+        assert data_api_client.update_service.called is False
+
+    def test_incorrect_framework_fails(self, data_api_client):
```

Comment:
You have a very similarly named test a few above - `test_edit_service_incorrect_framework` which both GET the same url. The test names are essentially identical but the difference is that one has a service and one doesnt (which then feels a bit weird because if it's the lack of service that's differing, why is the test name about incorrect frameworks).

I assume that you are actually testing correct things here but it might be worth thinking about putting these tests next to each other and renaming them to make it a bit clear what you mean by 'incorrect'